llvm/lib/CodeGen/MachineSink.cpp

```diff
@@ -100,6 +100,12 @@ static cl::opt<bool>
                                 "register spills"),
                        cl::init(false), cl::Hidden);
 
+static cl::opt<bool> AggressivelySinkInstsIntoCycle(
+    "aggressively-sink-insts-to-avoid-spills",
```

Comment:
Shouldn't include reasoning in the name. -aggressive-sink-insts-into-cycles? Not sure why we need this either, it seems like an incremental enhancement to SinkInstsIntoCycle

Reply:
I think we should keep it as opt-in to preserve the current behavior of sink-insts-to-avoid-spills

Reply:
But why? Don't do things out of fear of changing the behavior of something somewhere